test/manager_test.cpp

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "manager.hpp"
+
+#include <sdbusplus/async/context.hpp>
+
+#include <filesystem>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+class ManagerTest : public ::testing::Test
+{
+  protected:
+    static void SetUpTestCase()
+    {
+        char tmpdir[] = "/tmp/pdsCfgDirXXXXXX";
+        dataSyncCfgDir = mkdtemp(tmpdir);
+        std::filesystem::path dataSyncCfgFile{dataSyncCfgDir / "config.json"};
+        std::ofstream cfgFile(dataSyncCfgFile);
+
+        jsonData = R"(
+                {
+                    "Files": [
+                        {
+                            "Path": "/file/path/to/sync",
+                            "Description": "Parse test file",
+                            "SyncDirection": "Active2Passive",
+                            "SyncType": "Immediate"
+                        }
+                    ],
+                    "Directories": [
+                        {
+                            "Path": "/directory/path/to/sync",
+                            "Description": "Parse test directory",
+                            "SyncDirection": "Passive2Active",
+                            "SyncType": "Periodic",
+                            "Periodicity": "PT5M",
+                            "RetryAttempts": 1,
+                            "RetryInterval": "PT10M",
+                            "ExcludeFilesList": ["/directory/file/to/ignore"],
+                            "IncludeFilesList": ["/directory/file/to/consider"]
+                        }
+                    ]
+                }
+            )"_json;
+
+        cfgFile << jsonData;
+    }
+
+    static void TearDownTestCase()
+    {
+        std::filesystem::remove_all(dataSyncCfgDir);
+        std::filesystem::remove(dataSyncCfgDir);
```

Comment:
won't remove_all also remove the dir itself?

Reply:
Yeah, it’s not deleting. I’m a bit surprised too... it’s not like using `rm -rf`; it’s just meant to remove all its child files and subdirectories.